src/pkg/utils/utils.go

```diff
@@ -42,3 +44,31 @@ func UseLogFile() {
 		}
 	}
 }
+
+// MergeEnv merges two environment variable arrays,
+// replacing variables found in env2 with variables from env1
+// otherwise appending the variable from env1 to env2
+func MergeEnv(env1, env2 []string) []string {
```

Comment:
Great work on the `MergeEnv `function! I appreciate the effort to merge environment variables efficiently. I have a suggestion that could further enhance its performance and readability. Consider leveraging a map to reduce the complexity from `O(n*m)` to `O(n+m)`, where n and m are the lengths of `env1` and `env2`. This approach can significantly improve efficiency, especially with larger arrays. Here's how you might refactor it:

```go
func MergeEnv(env1, env2 []string) []string {
    envMap := make(map[string]string)
    var result []string

    // First, populate the map with env2's values for quick lookup.
    for _, s := range env2 {
        split := strings.SplitN(s, "=", 2)
        if len(split) == 2 {
            envMap[split[0]] = split[1]
        }
    }

    // Then, update the map with env1's values, effectively merging them.
    for _, s := range env1 {
        split := strings.SplitN(s, "=", 2)
        if len(split) == 2 {
            envMap[split[0]] = split[1]
        }
    }

    // Finally, reconstruct the environment array from the map.
    for key, value := range envMap {
        result = append(result, key+"="+value)
    }

    return result
}
```

This refactor optimizes the function and makes the code easier to understand and maintain. It's a suggestion :)